pallet/src/lib.rs

```diff
@@ -303,8 +303,22 @@ pub mod pallet {
     impl<T: Config> Did<T::AccountId, T::BlockNumber, <<T as Config>::Time as MomentTime>::Moment>
         for Pallet<T>
     {
-        fn is_owner(owner: &T::AccountId, did_account: &T::AccountId) -> Result<(), DidError> {
-            let id = (&owner, &did_account).using_encoded(blake2_256);
+        fn is_owner(
+            owner: &T::AccountId,
+            did_account: &T::AccountId,
+            name: &[u8],
+        ) -> Result<(), DidError> {
+            // Check for old key
+            let mut id = (&owner, &did_account).using_encoded(blake2_256);
+            if <OwnerStore<T>>::contains_key((&owner, &id)) {
+                // Add new key
+                let new_id = (&owner, &did_account, &name).using_encoded(blake2_256);
+                <OwnerStore<T>>::insert((&owner, &new_id), did_account);
+                // Remove old key
+                <OwnerStore<T>>::remove((&owner, &id));
+            }
+
+            id = (&owner, &did_account, &name).using_encoded(blake2_256);
```

Comment:
Could we reorg the flow? For example
1. If the new key gets, return
2. If found old key, go to upgrade on the fly
3. Return fails